src/components/Chat/Chat.jsx

```diff
@@ -1,39 +1,95 @@
-import React from 'react';
-// import { useSelector } from 'react-redux';
+import React, { useState, useEffect, useRef } from 'react';
+import { useSelector } from 'react-redux';
 import {
   Wrapper,
   Header,
   HeaderContent,
   MessageBlock,
   MessageBarWrapper,
   MessageBar,
+  ButtonAddFile,
   Input,
-  Button,
+  ButtonSendMessage,
 } from './ChatStyled';
 import Icons from '../Icons/Icons';
-// import { getCountryName } from 'redux-store/AuthOperations/selectors';
+import {
+  getCountryName,
+  getNumberOfParticipants,
+} from 'redux-store/AuthOperations/selectors';
 
 export default function Chat() {
-  // const countryName = useSelector(getCountryName);
+  const countryName = useSelector(getCountryName);
+  const participants = useSelector(getNumberOfParticipants);
+
+  const [messageBlockHeight, setMessageBlockHeight] = useState(0);
+  const [value, setValue] = useState('');
+
+  // const fileInput = React.createRef();
+  const textAreaRef = useRef(null);
+
+  const handleChange = e => {
+    setValue(e.target.value);
+    console.log(value);
+  };
+
+  const isInputNotEmpty = value.trim() !== '';
+
+  const updateMessageBlockHeight = () => {
+    if (textAreaRef.current) {
+      setMessageBlockHeight(textAreaRef.current.scrollHeight);
+    }
+  };
+
+  useEffect(() => {
+    textAreaRef.current.style.height = 'auto';
+    textAreaRef.current.style.height = textAreaRef.current.scrollHeight + 'px';
+    updateMessageBlockHeight();
+  }, [value]);
+
+  useEffect(() => {
+    const handleResize = () => {
+      updateMessageBlockHeight();
+    };
+  
+    window.addEventListener('resize', handleResize);
+  
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
```

Comment:
Line 37-59. Why do you need this? Try to use useEffect as less as possible.

Reply:
I need that this textarea for a message to increase when a text of the message bigger then one string so I founded only  this solution for this task. I need your advice what another way i can solution this.